crates/indexer-receipt/src/lib.rs

```diff
@@ -0,0 +1,175 @@
+        if receipts.is_empty() {
+            return Err(AggregationError::NoValidReceiptsForRavRequest);
+        }
+        let receipts: Vec<_> = receipts
+            .iter()
+            .map(|receipt| {
+                receipt
+                    .signed_receipt()
+                    .get_v1_receipt()
+                    .cloned()
+                    .ok_or(anyhow!("Receipt is not v1"))
+            })
+            .collect::<Result<_, _>>()
+            .map_err(AggregationError::Other)?;
+        let allocation_id = receipts[0].message.allocation_id;
+        tap_graph::ReceiptAggregateVoucher::aggregate_receipts(
+            allocation_id,
+            receipts.as_slice(),
+            previous_rav,
+        )
+    }
+}
+
+impl Aggregate<TapReceipt> for tap_graph::v2::ReceiptAggregateVoucher {
+    fn aggregate_receipts(
+        receipts: &[tap_core::receipt::ReceiptWithState<
+            tap_core::receipt::state::Checked,
+            TapReceipt,
+        >],
+        previous_rav: Option<tap_core::signed_message::Eip712SignedMessage<Self>>,
+    ) -> Result<Self, tap_core::receipt::rav::AggregationError> {
+        if receipts.is_empty() {
+            return Err(AggregationError::NoValidReceiptsForRavRequest);
+        }
+        let receipts: Vec<_> = receipts
+            .iter()
+            .map(|receipt| {
+                receipt
+                    .signed_receipt()
+                    .get_v2_receipt()
+                    .cloned()
+                    .ok_or(anyhow!("Receipt is not v2"))
+            })
+            .collect::<Result<_, _>>()
+            .map_err(AggregationError::Other)?;
+        let allocation_id = receipts[0].message.allocation_id;
+        let payer = receipts[0].message.payer;
+        let data_service = receipts[0].message.data_service;
+        let service_provider = receipts[0].message.service_provider;
+
+        tap_graph::v2::ReceiptAggregateVoucher::aggregate_receipts(
+            allocation_id,
+            payer,
+            data_service,
+            service_provider,
+            receipts.as_slice(),
+            previous_rav,
+        )
+    }
+}
+
+impl TapReceipt {
+    pub fn as_v1(self) -> Option<tap_graph::SignedReceipt> {
+        match self {
+            TapReceipt::V1(receipt) => Some(receipt),
+            _ => None,
+        }
+    }
+
+    pub fn as_v2(self) -> Option<tap_graph::v2::SignedReceipt> {
+        match self {
+            TapReceipt::V2(receipt) => Some(receipt),
+            _ => None,
+        }
+    }
+
+    pub fn get_v1_receipt(&self) -> Option<&tap_graph::SignedReceipt> {
+        match self {
+            TapReceipt::V1(receipt) => Some(receipt),
+            _ => None,
+        }
+    }
+
+    pub fn get_v2_receipt(&self) -> Option<&tap_graph::v2::SignedReceipt> {
+        match self {
+            TapReceipt::V2(receipt) => Some(receipt),
+            _ => None,
+        }
+    }
```

Comment:
Why is this necessary with the `v1` `v2` separation?

Reply:
just for Aggregation, it's especially important that we can only aggregate V1 Receipts into V1 Ravs and for V2 as well.

Reply:
Doesn't the enum handle that? I'm trying to understand how a V2 receipt could match as V1.

Reply:
How do you suggest doing it?

Reply:
I'm asking you, why you think this won't work:

```rust
pub fn get_receipt(&self) -> &tap_graph::v2::SignedReceipt {
        match self {
            TapReceipt::V2(receipt) => receipt,
            TapReceipt::V1(receipt) => receipt,
        }
    }
```

Maybe I'm missing something!

Reply:
Each one has a different type, the compiler would say `tap_graph::SignedReceipt` is not an instance of `tap_graph::v2::SignedReceipt`

Reply:
Right, I see that now, the `v1` having no similar label threw me. 

But do the `SignedReceipt` types not implement some shared behavior? But it doesn't have to look so confusing?